src/histogram.c

```diff
@@ -158,6 +158,18 @@ histogram_percentile_lower(struct histogram *hist, int pct)
 	return hist->max;
 }
 
+int64_t
+histogram_max(struct histogram *hist)
+{
+	for (int64_t i = hist->n_buckets - 1; i >= 0; i--) {
```

Comment:
struct histogram stores max itself, so why do you have to iterate over all buckets?

Reply:
max stored in struct histogram is not actual max (it is maximum for all time)